CONTRIBUTING.md

```diff
@@ -1,24 +1,178 @@
+  - [Prerequisites](#prerequisites)
+  - [Set up a development environment](#set-up-a-development-environment)
+      + [Setup a Conjur OSS Environment](#setup-a-conjur-oss-environment)
+      + [Setup Conjur identity on managed host](#setup-conjur-identity-on-managed-host)
+          - [Check Conjur identity](#check-conjur-identity)
+          - [Set up Conjur identity](#set-up-conjur-identity)
+          - [Set up Summon-Conjur](#set-up-summon-conjur)
+   - [Testing](#testing)
+   - [Releasing](#releasing)
 
-From a clean instance of main, perform the following actions to release a new version 
-of this plugin:
 
-- Update the version number in [`galaxy.yml`](galaxy.yml) and [`CHANGELOG.md`](CHANGELOG.md)
-    - Verify that all changes for this version in `CHANGELOG.md` are clear and accurate, 
-      and are followed by a link to their respective issue
-    - Create a PR with these changes
+ ## Prerequisites
+
+To start developing and testing using our development scripts ,
+the following tools need to be installed:
+
+1. [Git][get-git] to manage source code
+2. [Docker][get-docker] to manage dependencies and runtime environments
+3. [Docker Compose][get-docker-compose] to orchestrate Docker environments
+4. [Ruby][install-ruby] - native installation 
+5. [Python][Python] - for lookup plug-in
+
+
+[get-docker]: https://docs.docker.com/engine/installation
+[get-docker-compose]: https://docs.docker.com/compose/install
+[get-git]: https://git-scm.com/downloads
+[install-ruby]: https://www.ruby-lang.org/en/documentation/installation/
+[Python]: https://packaging.python.org/en/latest/tutorials/installing-packages/
+
+
+## Set up a development environment
+
+For developers, it is easy-to-use development environment, so that they can work on the collection, roles, plugins, and playbooks without needing to run the test suite. The repo scripts is in dev/ to setup and teardown this environment.
+This docker-compose dev environment is really useful, including a few different services:
+-  A Conjur Open Source instance
+-  An Ansible control node
+-  Managed nodes to push tasks to
+
+The `dev` directory contains a `docker-compose` file which creates a development
+environment with a database container (`pg`, short for *postgres*), and a
+`conjur` server container with source code mounted into the directory
+`/cyberark/dev/`.
+
+To use it:
+
+1. Install dependencies (as above)
+
+1. Start the container (and optional extensions):
+
+   ```sh-session
+   $ cd dev
+   $ ./start.sh
+   ...
+   root@f75015718049:/cyberark/dev/#
+   ```
+
+   Once the `start` script finishes, you're in a Bash shell inside the Conjur
+   server container.  To
+
+   After starting Conjur, your instance will be configured with the following:
+   * Account: `cucumber`
+   * User: `admin`
+   * Password: Run `cat conjur.identity` inside the container shell to display the current logged-in identity (which is also the password)
```

Comment:
Some of this section isn't accurate:

```md
1. Start the environment:

   ```sh-session
   $ cd dev
   $ ./start.sh
   ...
   PLAY RECAP *********************************************************************
   ansibleplugingtestingconjurhostidentity-test_app_centos-1 : ok=17 ...
   ansibleplugingtestingconjurhostidentity-test_app_centos-2 : ok=17 ...
   ansibleplugingtestingconjurhostidentity-test_app_ubuntu-1 : ok=16 ...
   ansibleplugingtestingconjurhostidentity-test_app_ubuntu-2 : ok=16 ...
   $
   \```

   Once the `start` script finishes, the environment's managed nodes have each been
   configured with a Conjur identity.

   After starting Conjur, your instance will be configured with the following:
   * Account: `cucumber`
   * User: `admin`
   * Password: Run `conjurctl role retrieve-key cucumber:user:admin` inside the Conjur container shell to retrieve the admin user API key (which is also the password)
```